app/models/server/models/LivechatVisitors.js

```diff
@@ -81,6 +82,22 @@ export class LivechatVisitors extends Base {
 		return this.update(query, update);
 	}
 
+	updateBatchLivechatDataByToken(token, fields) {
+		const query = { token };
+		const user = this.findOne(query, { fields: { livechatData: 1 } });
+		const updates = {};
+
+		fields.map((f) => {
+			if (!f.overwrite && user.livechatData && typeof user.livechatData[f.key] !== 'undefined') {
+				return null;
+			}
+			updates[`livechatData.${f.key}`] = f.value;
+			return null;
+		});
```

Comment:
```suggestion
		fields.forEach((f) => {
			if (!f.overwrite && user.livechatData && typeof user.livechatData[f.key] !== 'undefined') {
				return;
			}
			updates[`livechatData.${f.key}`] = f.value;
		});
```

Reply:
Since we're not converting/mapping an array here, we can use foreach loop instead